### ISSUE TITLE
Enable implicit dependencies for declared setup/teardown tasks

### ISSUE THREAD

**Issue by norm** [apache/airflow#29327]

**Comment by dstandish** [comment]
@norm can you clarify what you mean by implicit relationships?  i think what you mean is dependencies i.e. setup runs before normal tasks etc

**Comment by norm** [comment]
@dstandish Yes, sorry, dependencies. This means using `@setup`/`@teardown` on a task should create the dependency within the TG/DAG automatically rather than having to declare it `setup >> tasks >> teardown`.
